game/src/challenges/cutscene.rs

```diff
@@ -250,13 +248,11 @@ fn make_panel(
             }
             .margin_above(100),
             Widget::col(vec![
-                Widget::row(vec![prev, next]).centered_horiz(),
-                Btn::txt(
-                    "Skip cutscene",
-                    Text::from(Line("Skip cutscene").fg(Color::BLACK)),
-                )
-                .build_def(ctx, None)
-                .centered_horiz(),
+                Widget::row(vec![prev.margin_right(40), next]).centered_horiz(),
+                app.cs
```

Comment:
I made the buttons colors based on ColorScheme. I achieved this by having ColorScheme implement a new Button trait, and use it's fields to populate builder method arguments.

Overall, I'm satisfied with the flexibility of the new `ButtonBuilder`, but it's *quite* verbose.

One way to make it a little more concise would be to add a new top level item to app, like `app.btn`

```
struct App {
    ....
    // new field
    btn: AppButtons
}

// new struct
struct AppButtons {
    color_scheme: &ColorScheme
}
impl Button for AppButtons {
    fn primary_dark_text(&self, name: &str) -> {
        ...
    }
    fn secondary_dark_text(&self, name: &str) -> {
        ...
    }
    fn primary_light_text(&self, name: &str) -> {
        ...
    }
    fn secondary_light_text(&self, name: &str) -> {
        ...
    }
}
```

And hang an instance of `AppButtons` on App.

So then the below would be:

```
app.btn.secondary_dark_text("Skip cutscene")
    .build_def(ctx)
```

It's a tiny bit better.

We could also use terser names like: `app.bt.dark_text_prim("Skip cutscene")` `app.bt.dark_text_sec("Skip cutscene")`

I tend to prefer being more verbose, but maybe that's because I'm usually using an IDE these days. 🙃 

I feel like the bigger problem is the involvement of `app.cs`, but if I want them to be driven by the color scheme, I don't see a better way around that.

Any strong feelings on the matter @dabreegster ?

Reply:
I think I prefer `app.cs.btn_secondary_dark_text` to introducing the extra level of indirection. Also fine with the longer names.

> I feel like the bigger problem is the involvement of app.cs, but if I want them to be driven by the color scheme, I don't see a much better way...

I hit this tension when I was trying previously to divorce `App` and other non-widgetry things from creating widgets. Does it make sense to plumb this into `widgetry/src/style.rs`? The color scheme in `map_gui` sets that. Then the API might be something like the old `Btn::dark_text_primary(...)`, and all of the specific style stuff can only be fetched once that last call with `ctx` is plumbed in.